src/annotator/anchoring/pdf.js

```diff
@@ -448,55 +480,51 @@ export function anchor(root, selectors) {
  * @param {HTMLElement} root - The root element
  * @param {Range} range
  */
-export function describe(root, range) {
-  const normalizedRange = new BrowserRange(range).normalize();
+export async function describe(root, range) {
+  // "Shrink" the range so that the start and endpoints are at offsets within
+  // text nodes rather than any containing nodes.
+  try {
+    range = TextRange.fromRange(range).toRange();
+  } catch {
+    throw new Error('Selection does not contain text');
+  }
 
-  const startTextLayer = getNodeTextLayer(normalizedRange.start);
-  const endTextLayer = getNodeTextLayer(normalizedRange.end);
+  const startTextLayer = getNodeTextLayer(range.startContainer);
+  const endTextLayer = getNodeTextLayer(range.endContainer);
 
-  if (startTextLayer !== endTextLayer) {
-    return Promise.reject(
-      new Error('selecting across page breaks is not supported')
-    );
+  if (!startTextLayer || !endTextLayer) {
+    throw new Error('Selection is outside page text');
   }
 
-  const startRange = normalizedRange.limit(startTextLayer);
-  const endRange = normalizedRange.limit(endTextLayer);
-
-  if (!startRange || !endRange) {
-    return Promise.reject(new Error('range is outside text layer'));
+  if (startTextLayer !== endTextLayer) {
+    throw new Error('Selecting across page breaks is not supported');
```

Comment:
This is long-standing, but a pity.

Reply:
I don't see a reason why we can't fix this in future, but it is out of scope for this PR.